extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchModule.java

```diff
@@ -57,6 +59,11 @@ public void configure(final Binder binder)
     if (ComplexMetrics.getSerdeForType(MERGE_TYPE_NAME) == null) {
       ComplexMetrics.registerSerde(MERGE_TYPE_NAME, new HllSketchMergeComplexMetricSerde());
     }
+
+    if (binder != null) {
```

Comment:
It's possible to not need this if check by doing something like [this](https://github.com/apache/incubator-druid/pull/6950/files#diff-dbabe3c8d0b963a928eac3cef360880bR99) and using an injector

Reply:
I ended moving the complex metrics serde registration in these modules into `VisibleForTesting` static methods which are called from tests (so they don't need to call configure with a null binder).

Also tweaked the json mapper jackson module registration to use `CalciteTests.getJsonMapper()` instead of exposing `TestHelper.JSON_MAPPER`